src/models.rs

```diff
@@ -1131,3 +1134,170 @@ impl Metadata {
 
 /// Represent a String path instead of passing a PathBuf struct
 pub type StringPath = String;
+
+pub trait ToTerraformString {
+    fn to_terraform_format_string(&self) -> String;
+}
+
+pub trait ToHelmString {
+    fn to_helm_format_string(&self) -> String;
+}
+
+/// Represents a domain, just plain domain, no protocol.
+/// eq. `test.com`, `sub.test.com`
+#[derive(Clone)]
+pub struct Domain {
+    raw: String,
+    root_domain: String,
+}
+
+impl Domain {
+    pub fn new(raw: String) -> Self {
+        // TODO(benjaminch): This is very basic solution which doesn't take into account
+        // some edge cases such as: "test.co.uk" domains
+        let sep: &str = ".";
+        let items: Vec<String> = raw.split(sep).map(|e| e.to_string()).collect();
+        let items_count = raw.matches(sep).count() + 1;
+        let top_domain: String = match items_count > 2 {
+            true => items.iter().skip(items_count - 2).join("."),
```

Comment:
wasn't it more simple to do: format!("{}.{}", items[-2], items[-1])?

Reply:
There is for sure a better way / optimized from what I did,. but in your example, you are fetching array two times (it shouldn't be a big array, so not an issue).
Also, there is in my knowledge no such Python like negative indexing looping thing in Rust. But maybe I am wrong.